frontend/src/HomePage/SignHandlers/SignUp.jsx

```diff
@@ -19,19 +41,54 @@ function SignUp() {
             email: form.elements.email.value,
             password: form.elements.password.value
         };
-        try {
-            const response = await submitProfile(userData);
-            const data  = await response.json();
-            const id = data.userId
-            navigate(`/createprofile/${id}`);
-        } catch(error) {
-            console.error('sign up fail', error)
-            alert('Failed to Sign Up') // This will be replaced with more robust handling (e.g. being explicit about using an already registerd email)
+        const passwordData = isValidPassword(password)
+        if(passwordData[0]){
```

Comment:
I'm not in love with the callsite here doing `[0]`. Keying off an object would be nicer, imo. Its also worth doing a length check for passwordData so it doesn't crash in the future.

Reply:
I agree, it would be cleaner, I'll refactor that now and make a new commit to this PR addressing this.

Reply:
this should now be handled after #36fa947